src/EFCore.PG/Migrations/Internal/NpgsqlHistoryRepository.cs

```diff
@@ -127,6 +101,48 @@ SELECT 1 FROM pg_catalog.pg_class c
     protected override bool InterpretExistsResult(object? value)
         => (bool?)value == true;
 
+    bool IHistoryRepository.CreateIfNotExists()
+    {
+        // In PG, doing CREATE TABLE IF NOT EXISTS concurrently can result in a unique constraint violation
+        // (duplicate key value violates unique constraint "pg_type_typname_nsp_index"). We catch this and report that the table wasn't
+        // created.
+        try
+        {
+            return Dependencies.MigrationCommandExecutor.ExecuteNonQuery(
+                    GetCreateIfNotExistsCommands(), Dependencies.Connection, new MigrationExecutionState(), commitTransaction: true)
+                != 0;
+        }
+        catch (PostgresException e) when (e.SqlState == "23505")
```

Comment:
@AndriySvyryd note this annoying thing to deal with the PG problem of doing CREATE TABLE IF NOT EXISTS in parallel :/

Reply:
It's only a real issue when running test code that exercises these convoluted scenarios. I'd say that the correct thing here is to let it throw and handle it in the execution strategy used in tests